events/src/utils.py

```diff
@@ -313,6 +314,40 @@ def plugin_webhook(message):
     webhooks_pool.send(message["event"], message["ts"], message_json)
 
 
+class NamedTempOpen(object):
+    """
+        This class is used to create a temporary file which is then written to with contents.
+        The temp file is then persisted with the give name by calling os.rename().
+        This class is used to avoid the data loss or truncation in case of multiple processes
+        writing to the same file without the use of fcntl locks.
+
+        The temporary file is created in the cwd of the process.
```

Comment:
I think it would be better to create it in the same directory as the file we are creating. The cwd may be anything, and in some cases it could be inappropriate to create files there.

We could even create the temporary file as hidden by prefixing it with a `.`.

Reply:
Noted. Addressing this as we speak.

Reply:
You are still creating the temporary file in the working directory. I think this is not correct.

Reply:
Nope. Please see the changes in the most recent commit. This is an outdated one. The comment is also updated wrt the change.